### DISCUSSION TITLE
Possible to Set "Default Background" to be no background when viewing models?

### DISCUSSION THREAD

**Discussion by thegladscientist** [mozilla/hubs#3092] [General Questions & Discussion] [answered]
I have a question...is it possible to set the default background to be no background (show the scene) instead of being all black? I have some models that I would like to showcase for users coming in and out of a space (think museum style), and they are darker so do not show well on the default black background. 

While I have found out it is possible to change the background in the menus, I do not expect all users to be savvy enough to do this...is it possible to switch it so that "hide background" is the default when a user enters the room?

without background when viewing model
![2020-09-29 13_28_53-Chilly Authentic World _ Hubs by Mozilla](https://user-images.githubusercontent.com/2229204/94513129-dfcc0700-0258-11eb-83fd-091be9e7579f.png)

menu option to switch it
![2020-09-29 13_28_04-Chilly Authentic World _ Hubs by Mozilla](https://user-images.githubusercontent.com/2229204/94513125-de9ada00-0258-11eb-972f-70b1e1577310.png)

default black background when viewing model
![2020-09-29 13_31_10-Chilly Authentic World _ Hubs by Mozilla](https://user-images.githubusercontent.com/2229204/94513130-e0649d80-0258-11eb-99e0-052760898e5b.png)

**Comment by robinkwilson** [comment, marked as answer]
The way it works now, is it's hiding all the other objects in the scene when you inspect, so you're seeing the Threejs background.

@robertlong suggested you set a cubemap or another color as the ThreeJS scene background. https://threejs.org/docs/index.html#api/en/scenes/Scene.background

You can definitely do this with a [custom client](https://hubs.mozilla.com/docs/hubs-cloud-custom-clients.html) or (Note: I have NOT tried this on the <scene>) by a very "hacky way" with custom JS in the admin panel > Server Settings menu > Advanced tab. You might be able to use JavaScript to select the scene and update the background color with custom JavaScript.

Highly recommend using a custom client to do this, but the latter could be a lightweight (though duct taped) solution.

**Reply by thegladscientist** [reply]
Okay, I will try this and see how it goes, thank you for the response Robin!

**Reply by thegladscientist** [reply]
Hmmm...I actually don't have a server settings menu in my panel...does that require Hubs Cloud? Is there any way to do this without using Hubs Cloud?

**Reply by robinkwilson** [reply]
Unfortunately, Hubs Cloud is the path to update the Admin Panel + add your own code.

**Reply by thegladscientist** [reply]
Gotcha, was hoping there was some way to do it for regular hubs, but it's all good...thank you so much for your responses!

**Reply by thegladscientist** [reply]
@robin-k-wilson If I were to add to the codebase and make an option available to switch the background color...how would I go about exposing/integrating that into the menus for Hubs? Sorry if this info is somewhere else, but I am very curious about this feature. :)

**Reply by robinkwilson** [reply]
Make a Pull Request to the Hubs repo with the code to make it and/or feature request! PR would definitely be faster and SUPER appreciated. This would be a fantastic addition.
